src/basekernels/nn.jl

```diff
@@ -20,7 +20,35 @@ Bayesian neural network with erf (Error Function) as activation function.
 struct NeuralNetworkKernel <: Kernel end
 
 function (κ::NeuralNetworkKernel)(x, y)
-    return asin(dot(x, y) / sqrt((1 + sum(abs2, x)) * (1 + sum(abs2, y))))
+    return asin(dot(x, y) / sqrt((1 + sum(abs2.(x))) * (1 + sum(abs2.(y)))))
```

Comment:
This should not be needed since `Zygote.gradient(x -> sum(abs2, x), rand(5))` works fine. Revert it to
```suggestion
    return asin(dot(x, y) / sqrt((1 + sum(abs2, x)) * (1 + sum(abs2, y))))
```
?

Reply:
Yes I was testing if this was the issue. Forgot to revert it back.